### PR TITLE
Add MATE_DESKTOP_VERSION marco

### PR DESCRIPTION
**Pull request by zhuyaliang**
Add version macros for use by other applications

### CODE REVIEW

File: libmate-desktop/mate-desktop.h [hunk 1]
```diff
@@ -40,6 +40,7 @@ G_BEGIN_DECLS
         (MATE_MAJOR == (major) && MATE_MINOR == (minor) && \
         MATE_MICRO >= (micro)))
 
+#define MATE_DESKTOP_VERSION PACKAGE_VERSION
```
Comment (cwendling) [inline libmate-desktop/mate-desktop.h:43]: How would that work?  This relies on `config.h` defining `PACKAGE_VERSION`, which is the *built* project's version.  So inside libmate-desktop's build it'll be OK, but outside it won't.

Admittedly, existing `MATE_DESKTOP_CHECK_VERSION` isn't gonna work either outside of libmate-desktop, but there's no reason to add more non-working things :smiling_face_with_tear:

**Review by cwendling — CHANGES_REQUESTED** [review]
As mentioned below, this is not gonna work as expected.  Actually, this header is entirely non-functional as well.

So, first off, what is the purpose of this header, and this new constant especially?

We *can* fix this (generate this file from configure/meson so values like `MATE_MAJOR` & al. are hard-coded in it, and drop the *config.h* include), but given it's there since 2013 and it doesn't work, does anybody need any of this?

File: libmate-desktop/mate-desktop.h (hunk 1 above)
Comment (zhuyaliang) [reply libmate-desktop/mate-desktop.h:43]: My negligence, ```MATE_DESKTOP_CHECK_VERSION``` should be removed

Comment (cwendling) [reply libmate-desktop/mate-desktop.h:43]: Sure can do, but why do we need this?

Comment (zhuyaliang) [reply libmate-desktop/mate-desktop.h]: Provide a version interface for other external applications to obtain the ```mate-desktop``` version, such as the ```gdbus``` version interface of ```gnome-shell```  and the ```GNOME_DESKTOP_PLATFORM_VERSION``` macro provided by ```gnome -desktop```macro

**Comment by zhuyaliang** [comment]
@cwendling Please help me complete the remaining work, thank you

File: libmate-desktop/mate-desktop.h.in [hunk 2]
```diff
@@ -26,20 +26,6 @@
...
-
-#include <glib.h>
-
-G_BEGIN_DECLS
-
-#define MATE_DESKTOP_CHECK_VERSION(major,minor,micro) \
-        (MATE_MAJOR > (major) || \
-        (MATE_MAJOR == (major) && MATE_MINOR > (minor)) || \
-        (MATE_MAJOR == (major) && MATE_MINOR == (minor) && \
-        MATE_MICRO >= (micro)))
```
Comment (cwendling) [inline libmate-desktop/mate-desktop.h.in:41]: We can keep that if we want, just replace `MATE_MAJOR` & al. to it's actual value.

File: libmate-desktop/mate-desktop.h (hunk 1 above)
Comment (cwendling) [reply libmate-desktop/mate-desktop.h]: @zhuyaliang the autotools part is added.

For the rationale I'm not sure I understand the use case, but that's fine anyway, if you think it's useful, let's do it.
Should we make the `MATE_DESKTOP_CHECK_VERSION` work, or we still remove it?

File: libmate-desktop/mate-desktop.h.in [hunk 3]
```diff
@@ -26,20 +26,6 @@
...
-
-G_END_DECLS
+#define MATE_DESKTOP_VERSION @PACKAGE_VERSION@
```
[the remark this answers is not in the document]
Comment (zhuyaliang) [reply libmate-desktop/mate-desktop.h.in:29]: can refer to gnome
https://github.com/GNOME/gnome-desktop/blob/master/libgnome-desktop/gnome-desktop-version.h.in#L25

File: libmate-desktop/mate-desktop.h (hunk 1 above)
Comment (zhuyaliang) [reply libmate-desktop/mate-desktop.h]: I searched for the code and didn't find the use of  ```MATE_DESKTOP_CHECK_VERSION```,If you can add support in ```Makefile.am```, I can add support in ```meson```

File: libmate-desktop/mate-desktop.h.in (hunk 3 above)
[the remark this answers is not in the document]
Comment (cwendling) [reply libmate-desktop/mate-desktop.h.in:29]: Well, yes, but the value of `@GNOME_PLATFORM@` is actually a single integer (43, 44, etc.), whereas we have `1.27.1` or so, which is NOT a valid C identifier.

File: libmate-desktop/mate-desktop.h.in (hunk 2 above)
Comment (zhuyaliang) [reply libmate-desktop/mate-desktop.h.in:41]: ```MATE_DESKTOP_VERSION``` is sufficient to solve all the problem

File: libmate-desktop/mate-desktop.h.in (hunk 3 above)
[the remark this answers is not in the document]
Comment (zhuyaliang) [reply libmate-desktop/mate-desktop.h.in:29]: You're right

File: libmate-desktop/mate-desktop.h.in (hunk 2 above)
Comment (cwendling) [reply libmate-desktop/mate-desktop.h.in]: As you want, I just pushed a working version of `MATE_DESKTOP_CHECK_VERSION`, but feel free to drop it if it's useless, I don't mind.

Comment (zhuyaliang) [reply libmate-desktop/mate-desktop.h.in]: Thank you

File: libmate-desktop/mate-desktop.h.in (hunk 3 above)
[the remark this answers is not in the document]
Comment (zhuyaliang) [reply libmate-desktop/mate-desktop.h.in:35]: @cwendling Please help me fix this problem, my network is not good here

**Review by cwendling — APPROVED** [review]

**Comment by cwendling** [comment]
@zhuyaliang It should be good now, and I cleaned up a bit the commits (I can restore the previous version if you like it best though), it should be fine now.

**Pull request merged**